hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxySupport.java

```diff
@@ -215,6 +233,23 @@ public boolean removeIfSameInternal(Data key, Object value) {
         return true;
     }
 
+
+    private void unlock(Data key, VersionedValue versionedValue) {
+        try {
+            NodeEngine nodeEngine = getNodeEngine();
+            TxnUnlockOperation unlockOperation = new TxnUnlockOperation(name, key, versionedValue.version);
+            unlockOperation.setThreadId(ThreadUtil.getThreadId());
+            unlockOperation.setOwnerUuid(tx.getOwnerUuid());
+            int partitionId = nodeEngine.getPartitionService().getPartitionId(key);
+            Future<VersionedValue> future = nodeEngine.getOperationService()
+                    .invokeOnPartition(MapService.SERVICE_NAME, unlockOperation, partitionId);
+            future.get();
+            valueMap.remove(key);
```

Comment:
There is no need for this. All instances where the unlock is being called are made because the key does not exist in the map.